src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -639,14 +647,112 @@ void SyncManager::set_session_multiplexing(bool allowed)
 SyncClient& SyncManager::get_sync_client() const
 {
     util::CheckedLockGuard lock(m_mutex);
+    return do_get_sync_client();
+}
+
+SyncClient& SyncManager::do_get_sync_client() const
+{
     if (!m_sync_client)
-        m_sync_client = create_sync_client(); // Throws
+        m_sync_client = std::make_unique<SyncClient>(m_logger_ptr, m_config, weak_from_this()); // Throws
     return *m_sync_client;
 }
 
-std::unique_ptr<SyncClient> SyncManager::create_sync_client() const
+void SyncManager::set_sync_route(std::string sync_route)
+{
+    REALM_ASSERT(!sync_route.empty());
+    {
+        util::CheckedLockGuard lk(m_mutex);
+        m_sync_route = std::move(sync_route);
+    }
+    // Get a copy of the active sessions
+    auto sessions = get_all_sessions();
+    // Inform the active sessions that the sync route has been updated
+    for (auto& session : sessions) {
+        session->handle_location_updated();
+    }
+}
+
+void SyncManager::start_update_location(bool force_restart)
+{
+    util::CheckedLockGuard lk(m_mutex);
+    // If forcing a restart, cancel the current timer and restart the delay interval
+    if (force_restart) {
+        do_reset_update_location();
+    }
+    // If the location update hasn't been started yet, then start it now
+    if (!m_location_update_timer) {
+        do_update_location();
+    }
+}
+
+void SyncManager::do_reset_update_location()
+{
+    if (m_location_update_timer) {
+        m_location_update_timer->cancel();
+        m_location_update_timer.reset();
+    }
+    m_last_location_update_delay = 0;
+}
+
+void SyncManager::restart_location_update()
+{
+    util::CheckedLockGuard lk(m_mutex);
+    auto timer = std::move(m_location_update_timer);
```

Comment:
What does this do? Did you mean to call `do_reset_update_location()` to cancel any pending timers here?

Reply:
I was trying to "safely" replace the timer, but I reworked this logic so it is better to follow and if the timer needs to be canceled it takes place on the event loop via a post().